paasta_itests/setup_marathon_job.feature

```diff
@@ -18,6 +18,19 @@ Feature: setup_marathon_job can create a "complete" app
       And we run setup_marathon_job until it has 5 task(s)
      Then we should see the number of instances become 5
 
+  Scenario: marathon apps can be scaled up with at-risk hosts
+    Given a working paasta cluster
+      And a new healthy app to be deployed, with bounce strategy "crossover" and drain method "noop" and constraints [["hostname", "UNIQUE"]]
+      And an old app to be destroyed with constraints [["hostname", "UNIQUE"]]
+    When there are 2 old healthy tasks
+     And we mark the host "mesosslave" as at-risk
+     And setup_service is initiated
+    When there are 2 new healthy tasks
+     And setup_service is initiated
+     And we wait a bit for the old app to disappear
+    Then the old app should be gone
+     And the tasks on the host "mesosslave" should be drained
```

Comment:
I'm not convinced this test should pass consistently. This step is checking that there are no tasks on the `mesosslave` host, which doesn't match my understanding of what setup_marathon_job should do, at least not during the first run. If it's passing, I bet it's either because:
 - we're running setup_service twice after mesosslave is marked at-risk (first one scales up, second one scales down), and the steps in between are taking long enough that the scale up/down is finishing.
 - or because none of the original tasks were on mesosslave to begin with.

Let's do my suggestion where we pick one of the boxes that's at-risk, then after the next setup_service call, add `And there are 3 new healthy tasks`. Then, once we've run setup_service again, this last step asserting that mesosslave (or whatever we marked at-risk) is empty should pass.